### PR TITLE
Add removeEventListener definition to XRSession to prevent crash

### PR DESCRIPTION
**Pull request by chrisfromwork**
We have been seeing crashes on hololens when controllers connect and then disconnect. This crash was due to a missing remove event listener definition. This change adds a definitmion for that function.

### CODE REVIEW

File: Plugins/NativeXr/Source/NativeXr.cpp [hunk 1]
```diff
@@ -1706,7 +1707,7 @@ namespace Babylon
             JsRuntimeScheduler m_runtimeScheduler;
             uint32_t m_timestamp{0};
 
-            std::vector<std::pair<const std::string, Napi::FunctionReference>> m_eventNamesAndCallbacks{};
+            std::vector<std::pair<std::string, std::shared_ptr<Napi::FunctionReference>>> m_eventNamesAndCallbacks{};
```
Comment (bghgary) [inline Plugins/NativeXr/Source/NativeXr.cpp:1710]: why do you need to use shared_ptr?

Comment (bghgary) [inline Plugins/NativeXr/Source/NativeXr.cpp:1710]: Also, why remove const?

Comment (chrisfromwork) [reply Plugins/NativeXr/Source/NativeXr.cpp:1710]: if the string is declared as const, the "=" operator doesn't get defined for the std::pair. This breaks move operations used in remove_if/erase.

Comment (chrisfromwork) [reply Plugins/NativeXr/Source/NativeXr.cpp]: i think i was hitting issues with the const std::string, retesting this again it doesn't seem needed. It has been reverted.

**Comment by chrisfromwork** [comment]
Is there any update on this? What can I do to help get this checked in?

**Comment by bghgary** [comment]
Sorry, I'm running behind still.

**Comment by bghgary** [comment]
Merge from master and I'll merge.

**Pull request merged**